package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+          delete this._category;
+        }
+      },
+      //endregion
+
+      //region helpUrl property
+
+      // -> nonEmptyString, Optional, Inherited, Configurable, Localized?
+      // "" -> null conversion
+
+      _helpUrl: null,
+
+      /**
+       * Gets or sets an URL pointing to documentation associated with this type.
+       *
+       * Attempting to set to a non-string value type implicitly
+       * converts the value to a string before assignment.
+       *
+       * Setting to `undefined` causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where the attribute is `null`.
+       *
+       * Setting to `null` or to an empty string clears the attribute and sets it to `null`,
+       * ignoring any inherited value.
+       *
+       * @type {?nonEmptyString}
+       */
+      get helpUrl() {
+        return this._helpUrl;
+      },
+
+      set helpUrl(value) {
+        if(value === undefined) {
+          this._resetHelpUrl();
+        } else {
+          this._helpUrl = nonEmptyString(value);
+        }
+      },
+
+      _resetHelpUrl: function() {
+        if(this !== _type) {
+          delete this._helpUrl;
+        }
+      },
+      //endregion
+
+      //region isBrowsable property
+      // @type boolean
+      // -> boolean, Optional(true), Inherited, Configurable
+      // undefined or null -> resets
+
+      _isBrowsable: true,
+
+      /**
+       * Gets or sets the `isBrowsable` attribute of this type.
+       *
+       * Browsable types are exposed to the end user.
+       * Set this attribute to `false` to prevent exposing the type in a user interface.
+       *
+       * Setting to a {@link Nully} value causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where the attribute is `true`.
+       *
+       * @type {boolean}
+       */
+      get isBrowsable() {
+        return this._isBrowsable;
+      },
+
+      set isBrowsable(value) {
+        if(value == null) {
+          this._resetIsBrowsable();
+        } else {
+          this._isBrowsable = !!value;
+        }
+      },
+
+      _resetIsBrowsable: function() {
+        if(this !== _type) {
+          delete this._isBrowsable;
+        }
+      },
+      //endregion
+
+      //region isAdvanced property
+      // @type boolean
+      // -> boolean, Optional(false), Inherited, Configurable
+      // null || undefined -> reset
+      _isAdvanced: false,
+
+      /**
+       * Gets or sets the `isAdvanced` attribute of this type.
+       *
+       * Types with `isAdvanced` attributes set to `false` are typically immediately accessible to the user.
+       * An advanced type typically escapes the expected flow of utilization, yet it is
+       * sufficiently relevant to be shown in a user interface.
+       *
+       * Setting to a {@link Nully} value causes this attribute to use the inherited value,
```

Comment:
Suggest:

Setting this to a {@link Nully} value causes this attribute to use the inherited value,